pkg/medusa/requests.go

```diff
@@ -107,6 +112,56 @@ func (f *factory) NewRestoreRequest(ctx context.Context, restoreKey types.Namesp
 	return &req, nil, nil
 }
 
+func (f *factory) NewMedusaRestoreRequest(ctx context.Context, restoreKey types.NamespacedName) (*RestoreRequest, *ctrl.Result, error) {
+	restore := &api.MedusaRestoreJob{}
+	err := f.Get(ctx, restoreKey, restore)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, &ctrl.Result{}, nil
+		}
+		f.Log.Error(err, "Failed to get MedusaRestoreJob")
+		return nil, &ctrl.Result{RequeueAfter: 10 * time.Second}, err
+	}
+
+	backup := &api.MedusaBackup{}
```

Comment:
This part needs to be updated. I hit this error when I tried restoring to a new cluster. I had to manually create a sync task to avoid this.

Reply:
it's the process actually. Did you expect the sync to be done automatically?
Backups won't get synced unless you run a `sync` MedusaTask, then you'll have them created as MedusaBackup CR which can be restored.

Reply:
I think I misread the code last night and thought that the restore controller created the sync task as a preliminary step. It creates the prepare restore task. I'm good here.